Framework/Geometry/inc/MantidGeometry/Instrument/ParameterMap.h

```diff
@@ -381,8 +388,13 @@ class MANTID_GEOMETRY_DLL ParameterMap {
   /// Pointer to the DetectorInfo object. NULL unless the instrument is
   /// associated with an ExperimentInfo object.
   boost::shared_ptr<const Beamline::DetectorInfo> m_detectorInfo{nullptr};
+  /// Pointer to the ComponentInfo object. NULL unless the instrument is
+  /// associated with an ExperimentInfo object.
+  boost::shared_ptr<const Beamline::ComponentInfo> m_componentInfo{nullptr};
+  /// Component ids.
+  std::vector<Geometry::ComponentID> m_componentIds;
```

Comment:
I think that the storage mode here would be something to review as part of another issue. But those internals can be dealt with another time provided they are not causing big performance issues.